src/Aspire.Hosting.Testing/ResourceLogAggregatorBackgroundService.cs

```diff
@@ -0,0 +1,67 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.ApplicationModel;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+
+namespace Aspire.Hosting.Testing;
+
+/// <summary>
+/// A background service that aggregates resource logs and forwards them to the ILogger infrastructure.
+/// </summary>
+internal sealed class ResourceLogAggregatorBackgroundService(
```

Comment:
I had a thought this morning... should we put the request logs aggregator in Aspire.Hosting instead but just not enable it there by default, and then enable it in Aspire.Hosting.Testing, rather than it being exclusive to Aspire.Hosting.Testing? That way, if you want resource logs in the AppHost logs, you can do it easily, e.g. in tests that don't use the DistributedApplicationTestingBuilder.

Thoughts? @davidfowl @ReubenBond @mitchdenny

Reply:
I think this is worthwhile for the flexibility it adds.

Reply:
Sure, it's just "fresh" right now and trivial to do. But I can log a follow-up issue isntead.